mesonbuild/interpreter.py

```diff
@@ -2705,6 +2704,13 @@ def func_gettext(self, nodes, args, kwargs):
     def func_option(self, nodes, args, kwargs):
         raise InterpreterException('Tried to call option() in build description file. All options must be in the option file.')
 
+    def _handle_default_options(self, options: T.Union[str, T.List[str], T.Dict[str, str]]) -> 'T.OrderedDict[str, str]':
+        if isinstance(options, dict):
+            FeatureNew('Dictionary argument to default_options', '0.55.0').use(self.subproject)
+        else:
+            FeatureDeprecated('Array argument to default_options, use dict instead.', '0.50.0').use(self.subproject)
```

Comment:
I don't think we should deprecate, I don't see the problem in supporting both forms. No need to annoy users with deprecations warnings for no reason.

Reply:
Having two ways to do something in the long run just means more code to maintain and more code to break. The deprecation warning wont trigger until you set the `meson_version : >=0.55.0` anyway. But I could drop it I guess.

Reply:
IMHO, small and widely used things like this deprecation would be more burden than maintaining those 10 lines of code in Meson. I also find the list of strings syntax more compact and easier to write, so there is no clear benefit to use dict.

Reply:
This is a long standing and much used feature that would trigger many, many deprecation warnings. Since the implementation is so simple, I guess we can keep this working, at least for the moment. We can deprecate it later if needed.